backend/services/user.go

```diff
@@ -1,25 +1,65 @@
 package services
 
 import (
+	"backend/config"
 	"backend/models"
 	"backend/stores"
+
+	"golang.org/x/crypto/bcrypt"
 )
 
 type (
 	UserService interface {
 		GetUserById(id string) (*models.User, error)
-		// CreateUser(user *models.User) (string, error)
-		// UpdateUserById(user *models.User) (string, error)
-		// DeleteUser(id string) error
+		GetUserByUsername(username string) (*models.User, error)
+		CreateUser(creds *config.BasicAuth) (string, error)
+		DeleteUser(id string) error
 	}
 
 	userService struct {
 		stores *stores.Stores
 	}
 )
 
+func (s *userService) CreateUser(creds *config.BasicAuth) (string, error) {
+	encryptedPassword, err := bcrypt.GenerateFromPassword(
+		[]byte(creds.Password),
+		bcrypt.DefaultCost,
+	)
+	if err != nil {
+		return "", err
+	}
+	defaultAdmin := models.Admin{}
+	result := s.stores.DB.First(&defaultAdmin)
+
+	if result.Error != nil {
+		return "", err
+	}
+
+	user := models.User{
+		Username:     creds.Username,
+		PasswordHash: string(encryptedPassword),
+		Admin:        defaultAdmin,
+	}
+	user.PasswordHash = string(encryptedPassword)
+	user.Username = creds.Username
```

Comment:
[ask]
二重に代入しているので、必要なさそう？
```suggestion
	user := models.User{
		Username:     creds.Username,
		PasswordHash: string(encryptedPassword),
		Admin:        defaultAdmin,
	}
```

Reply:
ありがとうございます！修正完了しました。